### DISCUSSION TITLE
Mainsail v2.12.0 Release

### DISCUSSION THREAD

**Discussion by meteyou** [mainsail-crew/mainsail#1942] [Announcements]
Ahoi, sailors!

We have just released v2.12.0. This release includes two major features, many bug fixes, and minor improvements.

The first significant feature of this release is the ability to fully use Moonraker Sensors. This feature includes several components:
- Live view on the dashboard
- History totals
- History stats per print job

This feature can be used, for example, to log how much power a job has needed when a Tasmota (or similar MQTT solution) socket is installed and added to your Moonraker config. (this socket/component must be compatible with the [Moonraker sensor module](https://moonraker.readthedocs.io/en/latest/configuration/#sensor)).

The second prominent feature is the built-in themes. With version v2.12.0, there are now fixed integrated themes from large open-source communities:
- Klipper
- Voron Design
- VzBoT

The current community themes are just the beginning of this category. The Mainsail crew is only in direct contact with these maintainers, so we have started with these. If a maintainer of another project contacts us, more can follow.

We have also added themes from our vendor sponsors:
- BigTreeTech
- LDO Motion
- Multec GmbH
- Prusa Research
- YUMI

These themes should be a tiny bonus for our sponsors, without whom such a project would not be possible.

At last, I would like to mention a small feature contributed by @kevinkhill. With this feature, it is possible to save the file you are currently editing, or directly save and restart the component using hotkeys in the editor: [https://github.com/mainsail-crew/mainsail/pull/1902](https://github.com/mainsail-crew/mainsail/pull/1902).
Thank you very much for your contribution.

All other improvements and bugfixes can be found here: [https://github.com/mainsail-crew/mainsail/releases/tag/v2.12.0](https://github.com/mainsail-crew/mainsail/releases/tag/v2.12.0)

So, go ahead and hit refresh-button in your update manager and enjoy the new version of Mainsail!

**Happy printing**

PS: If you want to support the crew's work, consider the following:
- becoming a Patreon: [https://www.patreon.com/meteyou](https://www.patreon.com/meteyou)
- buy us a Ko-Fi: [https://ko-fi.com/mainsail](https://ko-fi.com/mainsail)

**Comment by kevinkhill** [comment]
Congrats on the release! I love this community and am humbled for the shout out. 🙏🥰

**Reply by meteyou** [reply]
thank you very much for your contribution!
